### PR TITLE
mm-api-ref-docs-adapter

### PR DESCRIPTION
**Pull request by mmiele**
Added ref documentation (WIP).

### CODE REVIEW

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py [hunk 1]
```diff
@@ -140,18 +180,27 @@ def __init__(self, settings: BotFrameworkAdapterSettings):
...
+        and is generally found in the `MicrosoftAppId` parameter in `config.py`.
+        :type bot_id: :class:`typing.str`
+        :param claims_identity: The bot claims identity
+        :type claims_identity: :class:`botframework.connector.auth.ClaimsIdentity`
+
+        :raises: It raises an argument null exception
+        
+        :return: A task that represents the work queued to execute
+        
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py [hunk 2]
```diff
@@ -176,12 +225,28 @@ def __init__(self, settings: BotFrameworkAdapterSettings):
...
+        :param logic: The logic to use for the creation of the conversation
+        :type logic: :class:`typing.Callable`  
+        :param conversation_parameters: The information to use to create the conversation
+        :type conversation_parameters:
+
+        :raises: It raises a generic exception error
+
+        :return: A task representing the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py [hunk 3]
```diff
@@ -354,9 +435,22 @@ def create_context(self, activity):
...
+        :param context: The context object for the turn
+        :type context: :class:`TurnContext'
+        :param activity: New replacement activity
+        :type activity: :class:`botbuilder.schema.Activity`
+
+        :raises: A generic exception error
+
+        :return: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]:
```suggestion
        .. note:: 
```

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py [hunk 4]
```diff
@@ -231,14 +296,26 @@ def __init__(self, settings: BotFrameworkAdapterSettings):
...
+        :param auth_header: The HTTP authentication header of the request
+        :type auth_header: :class:`typing.str`  
+        :param logic: The logic to execute at the end of the adapter's middleware pipeline.
+        :type logic: :class:`typing.Callable`  
+
+        :return: A task that represents the work queued to execute. If the activity type
+        was `Invoke` and the corresponding key (`channelId` + `activityId`) was found then 
+        an :class:`InvokeResponse` is returned; otherwise, `null` is returned.
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]:
```suggestion
        .. note:: 
```

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py [hunk 5]
```diff
@@ -373,9 +467,18 @@ def create_context(self, activity):
...
+        :param context: The context object for the turn
+        :type context: :class:`TurnContext'
+        :param reference: Conversation reference for the activity to delete
+        :type reference: :class:`botbuilder.schema.ConversationReference` 
+
+        :raises: A exception error
+
+        :return: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]:
```suggestion
        .. note:: 
```

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py (hunk 1 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py (hunk 2 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py (hunk 3 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py (hunk 4 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py (hunk 5 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/bot_framework_adapter.py]: Fixed

**Pull request merged**